spec/robots/dor_repo/ocr/fetch_files_spec.rb

```diff
@@ -0,0 +1,154 @@
+                label: 'image112.tif',
+                filename: 'image112.tif',
+                size: 123,
+                version: 1,
+                hasMimeType: 'image/tiff',
+                hasMessageDigests: [
+                  {
+                    type: 'sha1',
+                    digest: '5c9f6dc2ca4fd3329619b54a2c6f99a08c088444'
+                  },
+                  {
+                    type: 'md5',
+                    digest: 'ac440802bd590ce0899dafecc5a5ab1b'
+                  }
+                ],
+                access: {
+                  view: 'dark',
+                  download: 'none',
+                  controlledDigitalLending: false
+                },
+                administrative: {
+                  publish: false,
+                  sdrPreserve: true,
+                  shelve: false
+                }
+              }
+            ]
+          }
+        }
+      ],
+      hasMemberOrders: [],
+      isMemberOf: []
+    }
+  end
+
+  let(:dsa_object_client) do
+    instance_double(Dor::Services::Client::Object, find: cocina_model, update: true)
+  end
+
+  let(:pres_client) do
+    instance_double(Preservation::Client, objects: objects_client)
+  end
+
+  let(:objects_client) do
+    instance_double(Preservation::Client::Objects)
+  end
+
+  let(:workflow_client) do
+    instance_double(Dor::Workflow::Client, process: workflow_process, workflow_status: status)
+  end
+
+  let(:workflow_process) do
+    instance_double(Dor::Workflow::Response::Process, lane_id:, context:)
+  end
+
+  let(:lane_id) { 'lane1' }
+  let(:context) { { 'runOCR' => true } }
+  let(:status) { 'waiting' }
+
+  before do
+    allow(Dor::Services::Client).to receive(:object).and_return(dsa_object_client)
+    allow(Preservation::Client).to receive(:configure).and_return(pres_client)
+    allow(LyberCore::WorkflowClientFactory).to receive(:build).and_return(workflow_client)
+    allow(objects_client).to receive(:content) do |*args|
+      args.first.fetch(:on_data).call('CONTENT')
+    end
+  end
+
+  describe '#perform' do
+    subject(:perform) { test_perform(robot, druid) }
+
+    context 'with two image files' do
+      it 'calls gets cocina from DSA' do
+        perform
+        expect(dsa_object_client).to have_received(:find)
+      end
+
+      it 'configures preservation client' do
+        perform
+        expect(Preservation::Client).to have_received(:configure)
+      end
+
+      it 'calls the workflow service to get the context' do
+        perform
+        expect(workflow_client).to have_received(:process)
+      end
+
+      it 'gets the files' do
+        perform
+        expect(objects_client).to have_received(:content).twice
+        expect(File.exist?(File.join(Settings.sdr.abbyy_ticket_path, 'image111.tif'))).to be true
+        expect(File.exist?(File.join(Settings.sdr.abbyy_ticket_path, 'image112.tif'))).to be true
+      end
+    end
+  end
+end
```

Comment:
Note I didn't really test differently shaped Cocina since it seemed like that's the job of `spec/lib/dor/text_extraction/ocr_spec.rb` ?